src/func_cache.c

```diff
@@ -91,6 +84,19 @@ time_bucket_sort_transform(FuncExpr *func)
 	return (Expr *) copyObject(second);
 }
 
+static Expr *
+time_bucket_sort_transform(FuncExpr *func)
+{
+	Assert(list_length(func->args) >= 2);
+	/*
+	 * If period and offset are not constants we must not do the optimization
+	 */
+	if (!time_bucket_has_const_period_and_offset(func))
+		return (Expr *) func;
+
+	return time_bucket_gapfill_sort_transform(func);
```

Comment:
time_bucket_transform should not call time_bucket_gapfill_sort_transform it should be the other way around or separate pathes